sdks/java/core/src/main/java/org/apache/beam/sdk/options/ValueProvider.java

```diff
@@ -58,29 +58,24 @@ public void serialize(ValueProvider<?> value, JsonGenerator jgen, SerializerProv
   }
 
   static class Deserializer extends JsonDeserializer<ValueProvider<?>> {
+    private static final ObjectMapper MAPPER = new ObjectMapper();
```

Comment:
You shouldn't need another mapper, you should be able to get the json deserializer from the ctxt for the type that you want. This will then allow you to use the object mapper your already being executed part of.